homeassistant/components/telegram_bot/bot.py

```diff
@@ -1132,6 +1145,41 @@ async def load_data(
         )
 
 
+def _prepare_internal_url(hass: HomeAssistant, url: str) -> tuple[str, SSLContext]:
+    """Prepare internal Home Assistant URL with signed path.
+
+    Converts internal URLs (relative paths or HA instance URLs) to internal
+    URLs with signed authentication paths.
+    """
+    if hass.config.api is None:
+        raise HomeAssistantError(
+            "Home Assistant API not available",
+            translation_domain=DOMAIN,
+            translation_key="failed_to_load_url",
+            translation_placeholders={"error": "Home Assistant API not available"},
+        )
+
+    # Extract path from full URL or use directly if relative
+    path = str(YarlURL(url).relative()) if not url.startswith("/") else url
+
+    # Sign the path for authentication (5 minute expiry)
+    signed_path = async_sign_path(
+        hass, path, timedelta(minutes=5), use_content_user=True
+    )
+
+    internal_url = get_url(
+        hass,
+        allow_internal=True,
+        allow_ip=True,
+        prefer_external=False,
+        require_ssl=False,
+    )
+    internal_url = f"{internal_url}{signed_path}"
+
+    # Disable SSL verification (cert may not match the IP address)
+    return internal_url, get_default_no_verify_context()
```

Comment:
It is not necessarily clear to me that we should unilaterally disable ssl cert verification here. my impression is this is already a configuration option so unclear that we should override it or give the impression that we are verifying ssl certs but then not?